### PR TITLE
Fixes to test_onnx_export when saving input and output tensors

### PR DESCRIPTION
**Pull request by galagam**
- Allow saving i/o tensors when onnxruntime inference is skipped
- Support saving multiple outputs

### CODE REVIEW

File: tests/pytorch/test_onnx_export.py [hunk 1]
```diff
@@ -951,11 +965,11 @@ def test_export_multihead_attention(
...
     ).to(device='cuda')
-    do_export(model, inp, fname, use_fp8, input_names=input_names)
+    do_export(model, inp, fname, use_fp8, input_names=input_names, output_names=output_names)
+    infer_ort = precision != torch.float16  # temporarily skipping onnxrt inference due to input type mismatch bug
     if not use_fp8:
-        validate_result(fname, inp, model, atol=1e-3)
-    elif precision != torch.float16:
-        validate_result(fname, inp, model, atol=1e-2, is_fp8=use_fp8)
+        validate_result(fname, inp, model, atol=1e-3, input_names=input_names, output_names=output_names)
+    validate_result(fname, inp, model, atol=1e-2, is_fp8=use_fp8, input_names=input_names, output_names=output_names, infer_ort=infer_ort)
```
Comment (timmoon10) [inline tests/pytorch/test_onnx_export.py:972]: This this functional change intended? If not:
```suggestion
    else:
        validate_result(fname, inp, model, atol=1e-2, is_fp8=True, input_names=input_names, output_names=output_names, infer_ort=infer_ort)
```

File: tests/pytorch/test_onnx_export.py [hunk 2]
```diff
@@ -1013,10 +1027,10 @@ def test_export_transformer_layer(
         fuse_qkv_params=fuse_qkv_params,
         zero_centered_gamma=zero_centered_gamma).to(device='cuda')
     do_export(model, inp, fname, use_fp8)
+    infer_ort = precision != torch.float16  # temporarily skipping onnxrt inference due to input type mismatch bug
     if not use_fp8:
-        validate_result(fname, inp, model, atol=1e-3)
-    elif precision != torch.float16:
-        validate_result(fname, inp, model, atol=5e-1, is_fp8=use_fp8)
+        validate_result(fname, inp, model, atol=1e-3, input_names=input_names)
+    validate_result(fname, inp, model, atol=5e-1, is_fp8=use_fp8, input_names=input_names, infer_ort=infer_ort)
```
Comment (timmoon10) [inline tests/pytorch/test_onnx_export.py:1033]: If functional change is not intended:
```suggestion
    else:
        validate_result(fname, inp, model, atol=5e-1, is_fp8=True, input_names=input_names, infer_ort=infer_ort)
```

**Review by ksivaman — APPROVED** [review]

**Pull request merged**